src/backend/blockchain/tests/functional/test_routes.py

```diff
@@ -114,57 +139,66 @@ def test_add_wallet(test_client):
     response = test_client.post(url, data=json.dumps(data), headers=headers)
 
     assert response.status_code == 201
-    assert response.json['success'] == True
+    assert response.json['success']
 
     response = test_client.post(url, data=json.dumps(data), headers=headers)
 
     assert response.status_code == 400
     assert response.json['error'] == "wallet ID already exists"
 
+def test_add_wallet_incorrect_payload(test_client):
+    url = '/wallet/addWallet'
+    data = {'notumnetId': "fake_wallet_id"}
 
-def test_receiver_notpresent(test_client):
+    response = test_client.post(url, data=json.dumps(data), headers=headers)
+
+    assert response.status_code == 400
+    assert FailureReturnString.INCORRECT_PAYLOAD.value.encode() in response.data
+
+def test_sender_notpresent(test_client, test_transaction_payload):
+    test_transaction_payload['from'] = 'notpresent'
     url = '/wallet/createTransaction'
-    data = {'from': "fake_wallet_id",
-            'amount': 0,
-            'to': "another",
-            'timestamp': 0, "id": "abc", "signature": "123"}
 
-    response = test_client.post(url, data=json.dumps(data), headers=headers)
+    response = test_client.post(url, data=json.dumps(test_transaction_payload), headers=headers)
 
     assert response.status_code == 400
-    assert b'id' in response.data
+    assert b'no corresponding wallet for id' in response.data
```

Comment:
I might suggest having these return strings that we're checking for be put in variables as well. Then If they return strings change we only have to change one spot and not each individual occurrence.

Reply:
ah, i remember noting to change it, guess it missed these. Thanks for noticing!